diff --git a/EMULATOR.INI b/EMULATOR.INI
index 3c39c8b..b7e7d29 100644
--- a/EMULATOR.INI
+++ b/EMULATOR.INI
@@ -6,7 +6,7 @@
 ;ROMS subdirectory in the same folder than the exe is always scanned when searching for roms
 ;(remove the ; form Dir1)
 [RomDirs]
-;Dir1=c:\Model2\roms
+Dir1=z:\userdata\roms\model2
 
 
 [Renderer]
@@ -36,10 +36,10 @@ FSAA=0				;Enable full screen antialiasing in Direct3D
 
 
 ;These options are configured from menus so don't touch
-FullMode=0	
+FullMode=1	
 Sound=1
 Frameskip=-1
-AutoFull=0
+AutoFull=1
 Filter=20373472
 
 
